app/services/applied_add_ons/create_service.rb

```diff
@@ -2,74 +2,41 @@
 
 module AppliedAddOns
   class CreateService < BaseService
-    def create(**args)
-      @customer = Customer.find_by(
-        id: args[:customer_id],
-        organization_id: args[:organization_id],
-      )
-
-      @add_on = AddOn.find_by(
-        id: args[:add_on_id],
-        organization_id: args[:organization_id],
-      )
-
-      process_creation(
-        amount_cents: args[:amount_cents] || add_on&.amount_cents,
-        amount_currency: args[:amount_currency] || add_on&.amount_currency,
-      )
+    def self.call(...)
+      new(...).call
     end
 
-    def create_from_api(organization:, args:)
-      @customer = Customer.find_by(
-        external_id: args[:external_customer_id],
-        organization_id: organization.id,
-      )
-
-      @add_on = AddOn.find_by(
-        code: args[:add_on_code],
-        organization_id: organization.id,
-      )
+    def initialize(customer:, add_on:, amount_cents:, amount_currency:)
```

Comment:
What about using `params` instead of `amount_cents and amount_currency`?

If a new field is added in the future, we won't create a new parameter 😅